docs/eventing/sugar/README.md

```diff
@@ -0,0 +1,92 @@
+Knative Eventing Sugar Controller will react to special labels and annotations
+to produce or control eventing resources in a cluster or namespace.
+
+## Installing
+
+The following command installs the Eventing Sugar Controller:
+
+```bash
+kubectl apply --filename {{< artifact repo="eventing" file="eventing-sugar-controller.yaml" >}}
+```
+
+## Automatic Broker Creation
+
+One way to create a Broker is to manually apply a resource to a cluster using
+the default settings:
+
+```shell
+kubectl create -f - <<EOF
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: default
+  namespace: default
+EOF
+```
+
+There might be cases where automated Broker creation is desirable, such as on
+namespace creation, or on Trigger creation. The Sugar controller enables those
+use-cases:
+
+- When a Namespace is labeled with `eventing.knative.dev/injection=enabled`, the
+  controller will create a default broker in that namespace.
+- When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the
+  controller will create a Broker named by that Trigger in the Trigger's
+  Namespace.
+
+When a Broker is deleted and the above labels or annotations are in-use, the
+Sugar Controller will automatically recreate a default Broker.
+
+### Namespace Examples
+
+Creating a "default" Broker when creating a Namespace:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: example
+  labels:
+    eventing.knative.dev/injection: enabled
+EOF
+```
+
+To automatically create a Broker after a namespace exists, label the Namespace:
+
+```shell
+kubectl label namespace default eventing.knative.dev/injection=enabled
+```
+
+If the Broker named "default" already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+### Trigger Examples
+
+To create a Broker if none exists for the provided Broker name in the Namespace
```

Comment:
This seems like it's a useful pattern to allow defining an application without creating conflicts on the Broker resource if you have several applications sharing a Broker. Is it worth calling out that pattern?

Reply:
I am not sure I understand this comment, what is the suggestion?

Reply:
I think Evan is suggesting that you add a possible use case to help the reader understand, why would I use this?

IIUC, the use case is, I have multiple apps that will share a broker, and I want the broker to be created when any of those apps are created. BUT I don't want those apps to have to worry about keeping the broker definitions in sync in their yaml.